autofaiss/external/build.py

```diff
@@ -92,128 +99,250 @@ def get_estimated_construction_time_infos(nb_vectors: int, vec_dim: int, indent:
     return infos
 
 
-def create_index(
+def get_write_ids_df_to_parquet_fn(ids_root_dir: str) -> Callable[[pd.DataFrame, int], None]:
+    """Create function to write ids from Pandas dataframe to parquet"""
+
+    def _write_ids_df_to_parquet_fn(ids: pd.DataFrame, batch_id: int):
+        filename = f"part-{batch_id:08d}-{uuid.uuid1()}.parquet"
+        output_file = os.path.join(ids_root_dir, filename)  # type: ignore
+        with fsspec.open(output_file, "wb") as f:
+            logger.debug(f"Writing id DataFrame to file {output_file}")
+            ids.to_parquet(f, index=False)
+
+    return _write_ids_df_to_parquet_fn
+
+
+def get_optimize_index_fn(
     embedding_reader: EmbeddingReader,
     index_key: str,
-    metric_type: Union[str, int],
-    current_memory_available: str,
-    embedding_ids_df_handler: Optional[Callable[[pd.DataFrame, int], Any]] = None,
-    use_gpu: bool = False,
-    make_direct_map: bool = False,
-    distributed: Optional[str] = None,
-    temporary_indices_folder: str = "hdfs://root/tmp/distributed_autofaiss_indices",
-    nb_indices_to_keep: int = 1,
-    index_optimizer: Callable = None,
-) -> Tuple[Optional[faiss.Index], Dict[str, str]]:
-    """
-    Function that returns an index on the numpy arrays stored on disk in the embeddings_path path.
-    """
+    index_path: Optional[str],
+    index_infos_path: Optional[str],
+    use_gpu: bool,
+    save_on_disk: bool,
+    max_index_query_time_ms: float,
+    min_nearest_neighbors_to_retrieve: int,
+    index_param: Optional[str] = None,
+) -> Callable[[faiss.Index, str], Dict]:
+    """Create function to optimize index by choosing best hyperparameters and calculating metrics"""
+
+    def _optimize_index_fn(index: faiss.Index, index_suffix: str):
+        cur_index_path = make_path_absolute(index_path) + index_suffix if index_path else None
+        cur_index_infos_path = make_path_absolute(index_infos_path) + index_suffix if index_infos_path else None
+        if any(re.findall(r"OPQ\d+_\d+,IVF\d+_HNSW\d+,PQ\d+", index_key)):
+            set_search_hyperparameters(index, f"nprobe={64},efSearch={128},ht={2048}", use_gpu)
```

Comment:
These parameters are hardcoded?  And why would we do that for this specific case? I believe it is a remaining code after some test

Reply:
Ah ok no, it was in the previous code
It is very probably useless